pydatalab/src/pydatalab/routes/v0_1/info.py

```diff
@@ -143,3 +144,36 @@ def list_block_types():
             ).json()
         )
     )
+
+
+def get_all_models():
+    return Item.__subclasses__()
+
+
+@INFO.route("/info/types", methods=["GET"])
+def list_supported_types_schemas():
+    """Returns a dictionary of supported item types and their schemas."""
+    schemas = {cls.__name__.lower(): cls.schema() for cls in get_all_models()}
+    schemas["collections"] = Collection.schema()
+
+    return jsonify(schemas)
+
+
+for model_class in get_all_models():
+    model_name = model_class.__name__.lower()
+
+    def make_route(model_class):
+        @INFO.route(
+            f"/info/types/{model_name}", methods=["GET"], endpoint=f"get_{model_name}_schema"
+        )
+        def get_model_schema():
+            """Returns the JSON schema for the model."""
+            return jsonify(model_class.schema())
```

Comment:
It's just convention, but nesting this response under the keys `data->schema` would be good, then we can add any extra metadata or external links we want. We might want to introduce other fields for custom schemas, like `data->source`, or even just a simple list of properties (easier to access than within the JSONschema itself). It might be benefiical to grab the default metadata from the other info endpoint here.